File: movies-front/src/app/models/Evaluation.ts
import { Movie } from './Movie';
import { User } from './User.model';

export interface Evaluation {
  id?: number;
  name?: string; // Para compatibilidade
  authorName?: string; // Nome do autor da avaliação
  comment?: string;
  rating?: number; // 1 a 5 estrelas
  score?: number; // Para compatibilidade com backend
  createdAt?: Date;
  movie?: Movie;
  user?: User;
  movieId?: number;
  userId?: number;
}

File: movies-api/src/main/java/com/apispring/moviesapi/entities/Evaluation.java
package com.apispring.moviesapi.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(name = "evaluation")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder(setterPrefix = "set")
public class Evaluation {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private Long id;

    @Column
    private Integer score; // Rating de 1 a 5 estrelas

    @Column
    private String comment;

    @Column(name = "author_name")
    private String authorName; // Nome do autor da avaliação

    @ManyToOne
    @JoinColumn(name = "movie_id")
    private Movie movie;

}
